nixos/doc/manual/release-notes/rl-2009.xml

```diff
@@ -89,6 +89,14 @@ services.mysql.initialScript = pkgs.writeText "mariadb-init.sql" ''
       When MariaDB data directory is just upgraded (not initialized), the users are not created or modified.
     </para>
    </listitem>
+   <listitem>
+    <para>
+      MySQL server is now started with additional sandbox/hardening options. To allow reading the directory <literal>/home</literal> from MySQL (function LOAD DATA INFILE) use
+<programlisting>
+systemd.services.mysql.serviceConfig.ProtectHome = "read-only";
+</programlisting>
```

Comment:
Maybe something a little more explanatory like this:
```
MySQL server is now started with additional systemd sandbox/hardening options for better security. The PrivateTmp, ProtectHome, and ProtectSystem options may be problematic when MySQL is attempting to read from or write to your filesystem anywhere outside of its own state directory, for example when calling LOAD DATA INFILE or SELECT * INTO OUTFILE. In this scenario a variant of the following may be required:

<programlisting>
# allow MySQL to read from /home and /tmp directories when using LOAD DATA INFILE
systemd.services.mysql.serviceConfig.ProtectHome = lib.mkForce "read-only";
systemd.services.mysql.serviceConfig.PrivateTmp = lib.mkForce false;

# allow MySQL to write to /data when using SELECT * INTO OUTFILE, assuming the mysql user has write access to /data
systemd.services.mysql.serviceConfig.ReadWritePaths = [ "/data" ];
</programlisting>

<note>Carefully consider the security implications before making any such additions to your NixOS configuration files.</note>
```

Keep in mind my `<note>` tags are probably wrong. This documentation format isn't my strong point.

Reply:
Thanks, update release notes.